src/services/walktroughServices.js

```diff
@@ -0,0 +1,109 @@
+import { create, watch, currentUser, OpenShiftWatchEvents } from './openshiftServices';
+import {
+  cleanUsername,
+  findOpenshiftResource,
+  buildServiceInstanceResourceObj,
+  findOrCreateOpenshiftResource
+} from './middlewareServices';
+import * as osKinds from '../common/openshiftKinds';
+import { middlewareTypes } from '../redux/constants';
+import { FULFILLED_ACTION } from '../redux/helpers';
+import { DefaultServiceInstanceTransform } from '../common/serviceInstanceHelpers';
+
+/**
+ * Node messaging app transform, adds enmass credentials
+ */
+class NodeMessagingAppEnmassSiTransform {
+  isTransformable(siInfo) {
+    return siInfo.name === 'nodejs-messaging-work-queue-frontend';
+  }
+
+  transform(siInfo) {
+    const enmassCreds = {
+      MESSAGING_SERVICE_USER: 'username',
+      MESSAGING_SERVICE_PASSWORD: 'password'
+    };
+    const defaultTransform = new DefaultServiceInstanceTransform().transform(siInfo);
+    defaultTransform.spec.parameters = enmassCreds;
+    return defaultTransform;
+  }
+}
+
+const walkthroughs = {
+  one: {
+    namespaceSuffix: 'walkthrough-one',
+    services: ['spring-boot-rest-http-crud', 'nodejs-messaging-work-queue-frontend'],
+    transforms: [new DefaultServiceInstanceTransform()]
+  },
+  oneA: {
+    namespaceSuffix: 'walkthrough-one-a',
+    services: ['spring-boot-rest-http-crud', 'nodejs-messaging-work-queue-frontend'],
+    transforms: [new NodeMessagingAppEnmassSiTransform(), new DefaultServiceInstanceTransform()]
+  }
+};
+
+const prepareWalkthroughNamespace = (dispatch, walkthrough) => {
+  currentUser().then(user => {
+    const userNamespace = `${cleanUsername(user.username)}-${walkthrough.namespaceSuffix}`;
+    const namespaceObj = osKinds.buildNamespaceObj(userNamespace);
+    findOpenshiftResource(osKinds.projects, namespaceObj)
```

Comment:
I think the function `findOrCreateOpenshiftResource` does the check below for you.